crates/pipeline-types/src/config.rs

```diff
@@ -167,3 +172,31 @@ pub struct FormatConfig {
     #[schema(value_type = Object)]
     pub config: YamlValue,
 }
+
+#[derive(Debug, Clone, Eq, PartialEq, Default, Serialize, Deserialize, ToSchema)]
+pub struct ResourceConfig {
```

Comment:
I don't feel strongly about it, but I'd just make all these types here `usize` and call it a day... if everything is the same type makes it easier to compare and do calculation on them, otherwise I have to start worrying about overflow when I compose them.
Or if you have cpu_min or max and want to use this this to index an array at some point it has to be usize again

Reply:
These are passed to certain external types in a cluster manager (at which point they actually becomes numeric strings with units). I didn't want to import those types here, hence the selection.

Reply:
is it Mebibyte or Megabyte what we want?